### PR TITLE
Share pastes via email

### PR DESCRIPTION
**Pull request by nadinho**
This pull request adds a functionality to send each paste via Email if requested. 

To add this functionality, we

- used a Sendgrid API
- added a form with an input field for the email address and a button
- added a confirmation alert after sending the email

### CODE REVIEW

File: client/src/components/Approval.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import Alert from './Alert';
+
+function Approval(approved) {
+  //   const [approval, setApproval] = React.useState(false);
+  //   React.useEffect(() => {
+  //     setApproval(true);
+  //   });
+  if (approved === true) {
+    return <Alert>Email Successfully Sent 🎁</Alert>;
```
Comment (Lenkly) [inline client/src/components/Approval.js]: I would have used the Confirmation component, as the Alert has been used for warnings so far. (The alert has a red border whereas the confirmation has a green one.)

**Review by Lenkly — COMMENTED** [review]
Have a look at my comment ;)

File: client/src/pages/Paste.js [hunk 2]
```diff
@@ -38,6 +65,14 @@ function Paste({ match }) {
...
           <Author name={paste.author} />
           <PasteArea>{paste.value}</PasteArea>
+          <EmailInput
+            placeholder="Enter the email of your recipient"
+            value={inputValue}
+            type="email"
+            onChange={event => setInputValue(event.target.value)}
+          ></EmailInput>
+          <SendEmailButton onClick={handleClickEvent}></SendEmailButton>
+          <div>{Approval(approval)}</div>
```
Comment (hauptdigital) [inline client/src/pages/Paste.js]: The email type check does not work. We were able to get a "Email successfully send" without a valid email entry (i.e. "test@"). 

Maybe include an additional check for correct email syntax.

File: client/src/pages/Paste.js [hunk 3]
```diff
@@ -38,6 +65,14 @@ function Paste({ match }) {
           </CreatedAt>
           <Author name={paste.author} />
           <PasteArea>{paste.value}</PasteArea>
+          <EmailInput
+            placeholder="Enter the email of your recipient"
+            value={inputValue}
+            type="email"
+            onChange={event => setInputValue(event.target.value)}
+          ></EmailInput>
+          <SendEmailButton onClick={handleClickEvent}></SendEmailButton>
```
Comment (hauptdigital) [inline client/src/pages/Paste.js]: Multiple clicks on the button resulted in multiple emails. Maybe you can add a functionality to prevent mutliple clicks like a status "disabled" or a success state for the button so it is now clickable anymore. If input value changes, it might be useful to remove the disabled state to allow another share via email.

File: client/src/components/Approval.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import Alert from './Alert';
+
+function Approval(approved) {
+  //   const [approval, setApproval] = React.useState(false);
+  //   React.useEffect(() => {
+  //     setApproval(true);
+  //   });
```
Comment (hauptdigital) [inline client/src/components/Approval.js]: I think this is not needed anymore 😺

File: client/src/components/Approval.js (hunk 1 above)
Comment (nadinho) [reply client/src/components/Approval.js]: Thank you, we changed that :)

**Review by hauptdigital — CHANGES_REQUESTED** [review]
We added three comments to the code. Please have a look 👍 

https://github.com/lmachens/copypasta/pull/35/files#r399998349
https://github.com/lmachens/copypasta/pull/35/files#r399998349
https://github.com/lmachens/copypasta/pull/35/files#r399995227

File: client/src/pages/Paste.js (hunk 2 above)
Comment (nadinho) [reply client/src/pages/Paste.js]: Yeah, we are currently working on that problem. Thx :)

Comment (nadinho) [reply client/src/pages/Paste.js]: I fixed the email validation with adding a form element.  You'll still get an  "Email successfully send"  because for now it's just an alert.

File: client/src/components/Approval.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import Confirmation from './Confirmation';
+
+function Approval(approved) {
```
Comment (steinberggit) [inline client/src/components/Approval.js]: Why is the name of function in UpperCamelCase?

File: client/src/components/Checkbox.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin: 10px;
+  color: ${props => (props.disabled ? '#666' : 'inherit')};
+`;
+
+function Checkbox({ disabled, onChange, checked, label }) {
+  return (
+    <Label disabled={disabled}>
```
Comment (steinberggit) [inline client/src/components/Checkbox.js]: There exists no `disabled` attribute for the html element `label`

File: client/src/components/PasteBody.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      <PasteArea>{paste.value}</PasteArea>
+      {!embedded && paste.isEmbeddable && <EmbedButton pasteId={pasteId} />}
+      <EmailInput
+      // placeholder="Enter the email of your recipient"
+      // value={mailInputValue}
+      // type="email"
+      // onChange={event => setMailInputValue(event.target.value)}
+      ></EmailInput>
+      <SendEmailButton></SendEmailButton>
+      {/* <div>{Approval(approval)}</div> */}
```
Comment (steinberggit) [inline client/src/components/PasteBody.js]: ?

File: client/src/api/pastes.js [hunk 8]
```diff
@@ -42,3 +55,34 @@ export function addPastaPoint(pasteId) {
...
+      }
+    ],
+    content: [{ type: 'text/html', value: `${pasteId}` }],
+
+    from: { email: 'copypaste@gmx.de', name: 'CopyPasta 🍝' },
+
+    reply_to: { email: 'copypaste@gmx.de', name: 'CopyPasta 🍜' }
+  };
+
+  // setApproval(true);
```
Comment (steinberggit) [inline client/src/api/pastes.js]: Description why prevent execution

File: client/package.json [hunk 9]
```diff
@@ -9,7 +9,7 @@
     "react": "^16.10.1",
     "react-dom": "^16.10.1",
     "react-router-dom": "^5.1.2",
-    "react-scripts": "3.1.2"
+    "react-scripts": "^3.4.1"
```
Comment (lmachens) [inline client/package.json]: This change is not related to the pull request

File: client/src/components/Approval.js (hunk 5 above)
Comment (lmachens) [reply client/src/components/Approval.js]: This doesn't look like a React Component

File: client/src/components/PasteBody.js (hunk 7 above)
Comment (lmachens) [reply client/src/components/PasteBody.js]: `<Approval approved={approved} />` would be the correct syntax

**Review by lmachens — CHANGES_REQUESTED** [review]
